tests/classes/test-class-amp-admanager.php

```diff
@@ -0,0 +1,564 @@
+	 */
+	public function test_get_ads() {
+
+		// Mobile Ads.
+		$expected_output = '<amp-ad width="300" height="250" media="(max-width: 499px)" type="doubleclick" data-slot="/123456789/AMP_ADTest" json=\'{&quot;targeting&quot;:{&quot;contentType&quot;:&quot;&quot;,&quot;siteDomain&quot;:&quot;example.org&quot;,&quot;adId&quot;:&quot;AMP_ADTest&quot;}}\' data-multi-size="300x250" data-multi-size-validation="false" layout="fixed" data-loading-strategy="prefer-viewability-over-views" data-enable-refresh=></amp-ad>';
+		$ad_attr         = [
+			'ad-unit'      => 'AMP_ADTest',
+			'mobile-sizes' => '300x250',
+			'layout'       => 'fixed',
+		];
+		$output          = AMP_AdManager::get_ads( $ad_attr );
+
+		$this->assertNotEmpty( $output );
+		$this->assertEquals( $expected_output, $output );
+
+		// Tablet Ads.
+		$expected_output = '<amp-ad width="336" height="280" media="(min-width: 500px) and (max-width: 799px)" type="doubleclick" data-slot="/123456789/AMP_ADTest" json=\'{&quot;targeting&quot;:{&quot;contentType&quot;:&quot;&quot;,&quot;siteDomain&quot;:&quot;example.org&quot;,&quot;adId&quot;:&quot;AMP_ADTest&quot;}}\' data-multi-size="336x280" data-multi-size-validation="false" layout="fixed" data-loading-strategy="prefer-viewability-over-views" data-enable-refresh=></amp-ad>';
+		$ad_attr         = [
+			'ad-unit'      => 'AMP_ADTest',
+			'tablet-sizes' => '336x280',
+			'layout'       => 'fixed',
+		];
+		$output          = AMP_AdManager::get_ads( $ad_attr );
+
+		$this->assertNotEmpty( $output );
+		$this->assertEquals( $expected_output, $output );
+
+		// Desktop ads.
+		$expected_output = '<amp-ad width="970" height="250" media="(min-width: 800px)" type="doubleclick" data-slot="/123456789/AMP_ADTest" json=\'{&quot;targeting&quot;:{&quot;contentType&quot;:&quot;&quot;,&quot;siteDomain&quot;:&quot;example.org&quot;,&quot;adId&quot;:&quot;AMP_ADTest&quot;}}\' data-multi-size="970x250" data-multi-size-validation="false" layout="responsive" data-loading-strategy="prefer-viewability-over-views" data-enable-refresh=></amp-ad>';
+		$ad_attr         = [
+			'ad-unit' => 'AMP_ADTest',
+			'sizes'   => '970x250',
+		];
+		$output          = AMP_AdManager::get_ads( $ad_attr );
+
+		$this->assertNotEmpty( $output );
+		$this->assertEquals( $expected_output, $output );
+
+		// Test echo.
+		$output_echo = Utility::buffer_and_return( 'AMP_AdManager\AMP_AdManager::get_ads', [ $ad_attr, true ] );
+
+		$this->assertNotEmpty( $output_echo );
+		$this->assertEquals( $expected_output, $output_echo );
+
+	}
+
+	/**
+	 * Tests `filter_breakpoints` function.
+	 *
+	 * @covers ::filter_breakpoints
+	 *
+	 * @return void
+	 */
+	public function test_filter_breakpoints() {
+
+		$ad_attr = [
+			'ad-unit' => 'AMP_ADTest',
+			'sizes'   => '320x50',
+			'layout'  => 'fixed',
+		];
+
+		$output = Utility::invoke_method( $this->_instance, 'filter_breakpoints', [ $ad_attr['sizes'] ] );
+
+		$this->assertNotEmpty( $output );
+		$this->assertNotEmpty( $output );
```

Comment:
Duplicate assertion. I see the same assertion duplicated on multiple places. Please remove duplicate once from all the places.